src/disk_io_thread.cpp

```diff
@@ -1123,6 +1131,24 @@ namespace libtorrent
 
 		m_stats_counters.inc_stats_counter(counters::num_running_disk_jobs, -1);
 
+		mutex::scoped_lock l(m_cache_mutex);
+		int check_ret = check_cache_level(l, completed_jobs);
+		l.unlock();
+
+		if (check_ret == retry_job)
+		{
+			// we only need to post a job if there are no other jobs in the queue
+			// otherwise check_cache_level will be called again after the next job
```

Comment:
I'll look into this. There is at least one other path involving the block reference counts which can cause `check_cache_level` not to be able to evict that I'll have to handle. Admittedly I haven't fully investigated it, when I saw it I decided to go with the catch-all and just re-check `num_to_evict` at the end of `check_cache_level` for maximum robustness.